plugins/git-release-manager/src/api/GitReleaseApiClient.ts

```diff
@@ -0,0 +1,814 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ConfigApi, OAuthApi } from '@backstage/core';
+import { Octokit } from '@octokit/rest';
+import { readGitHubIntegrationConfigs } from '@backstage/integration';
+
+import { DISABLE_CACHE } from '../constants/constants';
+import { Project } from '../contexts/ProjectContext';
+import { UnboxArray, UnboxReturnedPromise } from '../types/helpers';
+
+export class GitReleaseApiClient implements GitReleaseApi {
+  private readonly githubAuthApi: OAuthApi;
+  private readonly baseUrl: string;
+  readonly host: string;
+
+  constructor({
+    configApi,
+    githubAuthApi,
+  }: {
+    configApi: ConfigApi;
+    githubAuthApi: OAuthApi;
+  }) {
+    this.githubAuthApi = githubAuthApi;
+
+    const githubIntegrationConfig = this.getGithubIntegrationConfig({
+      configApi,
+    });
+
+    this.host = githubIntegrationConfig?.host ?? 'github.com';
+    this.baseUrl =
+      githubIntegrationConfig?.apiBaseUrl ?? 'https://api.github.com';
+  }
+
+  private getGithubIntegrationConfig({ configApi }: { configApi: ConfigApi }) {
+    const configs = readGitHubIntegrationConfigs(
+      configApi.getOptionalConfigArray('integrations.github') ?? [],
+    );
+
+    const githubIntegrationEnterpriseConfig = configs.find(v =>
+      v.host.startsWith('ghe.'),
+    );
+    const githubIntegrationConfig = configs.find(v => v.host === 'github.com');
+
+    // Prioritize enterprise configs if available
+    return githubIntegrationEnterpriseConfig ?? githubIntegrationConfig;
+  }
+
+  private async getOctokit() {
+    const token = await this.githubAuthApi.getAccessToken(['repo']);
+
+    return {
+      octokit: new Octokit({
+        auth: token,
+        baseUrl: this.baseUrl,
+      }),
+    };
+  }
+
+  public getHost: GitReleaseApi['getHost'] = () => {
```

Comment:
> Do you need to specify the return types like this? TS should be warning you anyway if you return something that breaks the interface contract.

Not for methods without arguments, but for completeness I type them as well.

This is the output otherwise:

![Screen Shot 2021-05-05 at 17 01 46](https://user-images.githubusercontent.com/5392475/117163056-94b43c80-adc3-11eb-8298-5d7b07ce972d.png)

> Another note - matter of taste, but these should probably be on the form public getHost() { instead of as arrow functions

I agree that it's nicer, but it makes typing the method more difficult 🙁